port/unix/omrsysinfo.c

```diff
@@ -6775,13 +6902,49 @@ omrsysinfo_cgroup_subsystem_iterator_next(struct OMRPortLibrary *portLibrary, st
 		 */
 		size_t len = strlen(metricElement->value);
 		metricElement->units = currentElement->metricUnit;
-		if ((len > 0) && (metricElement->value[len-1] == '\n')) {
+		if ((len > 1) && (metricElement->value[len-1] == '\n')) {
```

Comment:
Is this because it's impossible for `len` to ever be `1`?

Reply:
len will be 1 when the file contains just a newline. This code removes a newline from the end of a string returned from fgets and I didn't want it to remove a sole newline so I changed it to ignore sole newlines and instead check for it later: `if ('\n' == metricElement->value[0]) {`
Otherwise this check would have had to check for '\0' which is a little less intuitive